src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/EETypeNode.cs

```diff
@@ -114,7 +114,14 @@ public override bool ShouldSkipEmittingObjectNode(NodeFactory factory)
 
         public virtual ISymbolNode NodeForLinkage(NodeFactory factory)
         {
-            return factory.NecessaryTypeSymbol(_type);
+            if (ConstructedEETypeNode.CreationAllowed(_type))
```

Comment:
I am not able to figure out why we need this change. Could you please shed some light on this?